core/src/providers/anthropic.rs

```diff
@@ -79,7 +81,119 @@ impl AnthropicLLM {
             .parse::<Uri>()?)
     }
 
-    async fn streamed_completion(
+    fn chat_prompt(&self, messages: &Vec<ChatMessage>) -> String {
+        let mut prompt = messages
+            .iter()
+            .map(|cm| -> String {
+                format!(
+                    "\n\n{}: {}",
+                    match cm.role {
+                        ChatMessageRole::System => "Assistant",
+                        ChatMessageRole::Assistant => "Assistant",
+                        ChatMessageRole::User => "Human",
+                    },
+                    cm.content
+                )
+            })
+            .collect::<Vec<_>>()
+            .join("");
+
+        prompt = format!("{}\n\nHuman:", prompt);
+
+        return prompt;
+    }
+
+    async fn chat_completion(
+        &self,
+        messages: &Vec<ChatMessage>,
+        temperature: f32,
+        top_p: f32,
+        stop: &Vec<String>,
+        max_tokens: Option<i32>,
+    ) -> Result<LLMChatGeneration> {
+        assert!(self.api_key.is_some());
+
+        let prompt = self.chat_prompt(messages);
+        let mut stop_tokens = stop.clone();
+        stop_tokens.push(String::from("\n\nHuman:"));
+        stop_tokens.push(String::from("\n\nAssistant:"));
+
+        let response = self
+            .completion(
+                self.api_key.clone().unwrap(),
+                &prompt,
+                match max_tokens {
+                    Some(m) => m,
+                    None => 256,
+                },
+                temperature,
+                top_p,
+                // @todo(aric): I don't understand the meaning
```

Comment:
This is to only use the top_k logprobs to sample instaed of the full distribution. It is optional and defaults to -1.
@fontanierh incorrectly mapped it to top_logprobs which is meant to controls the number of logprobs returned when they are returned. 

We shoud:
- make top_k optional in bot completion and streamed_completion
- do not map top_logprobs to it in existing code (use None) (and just ignore it, anthropic does not return logprobs right?)
- do not map it either here (use None)

Reply:
`top_k` will always be None as we don't support it in the `LLM` or `Chat` interface right now.